src/Chapter2/GeneralizedFibo.hs

```diff
@@ -0,0 +1,14 @@
+module Chapter2.GeneralizedFibo where
+
+import Data.List
+
+fiboList :: Int -> [Int]
+fiboList n= listOfFibo ( foldl' (  \(x,y,fiboList) i -> ( y, x+y ,   (fiboList++[x+y])  ) ) (0,1,[])  [1..n*(n+10)] )
```

Comment:
why are you doing `n*(n+10)`? Use a combination of higher-order functions and tail-recursion to solve this problem. You have already solved this using tail-recursing in http://www.vacationlabs.com/haskell/basic-types-and-functions.html#even-fibonacci-numbers (second technique of solving this problem). First write the code for the following using that technique, then generalize it by writing your own higher-order function:

- Write a function to return the **first N** terms of the fibonacci series that are even
- Write a function to return the **first N** terms of the fibonacci series that are multiples of 3
- Write a function to return the **first N** terms of the fibonacci series that are multiples of 5